src/reactor.js

```diff
@@ -26,9 +27,13 @@ class Reactor {
   constructor(config = {}) {
     const debug = !!config.debug
     const baseOptions = debug ? DEBUG_OPTIONS : PROD_OPTIONS
+    // if defined, merge the custom implementation over the noop logger to avoid undefined lookups,
+    // otherwise, just use the built-in console group logger
+    const debugLogger = config.logger ? extend({}, NoopLogger, config.logger) : ConsoleGroupLogger
     const initialReactorState = new ReactorState({
       debug: debug,
       cache: config.cache || DefaultCache(),
+      logger: debug ? debugLogger : NoopLogger,
```

Comment:
The way the current code is written it's not possible to have a logger when `debug` is falsy.

It'd be nice to be able to do:

```js
new Nuclear.Reactor({
  logger: MyLogger
})
```

A little context, the `debug` option is very old and existing before specific logging options were available, so I kept it around to maintain backwards compatibility.

Ideally, regardless of the value of `debug` if you pass in a `logger` option it should respect that.

```js
let logger = config.logger ? extend({}, NoopLogger, config.logger) : NoopLogger
if (!config.logger && debug) {
  logger = ConsoleGroupLogger
}
```

I believe that logic preserves the use cases I mentioned above.